datalab/datalab_session/tests/test_operations.py

```diff
@@ -0,0 +1,196 @@
+    def test_description(self):
+        self.assertEqual(self.data_operation.description(), 'Testing class for DataOperation')
+    
+    def test_wizard_description(self):
+        wizard_description = {
+            'name': 'SampleDataOperation',
+            'description': 'Testing class for DataOperation',
+            'category': 'test',
+            'inputs': {
+                'input_files': {
+                    'name': 'Input Files',
+                    'description': 'The input files to operate on',
+                    'type': 'file',
+                    'minimum': 1,
+                    'maximum': 999
+                },
+                'input_number': {
+                    'name': 'Input Number',
+                    'description': 'The input number to operate on',
+                    'type': 'number',
+                    'minimum': 1,
+                    'maximum': 999
+                },
+                'input_string': {
+                    'name': 'Input String',
+                    'description': 'The input string to operate on',
+                    'type': 'string',
+                    'minimum': 1,
+                    'maximum': 999
+                }
+            }
+        }
+        self.assertEqual(self.data_operation.wizard_description(), wizard_description)
+    
+    def test_operate(self):
+        self.data_operation.operate()
+        self.assertEqual(self.data_operation.get_percent_completion(), 1.0)
+        self.assertEqual(self.data_operation.get_status(), 'COMPLETED')
+        self.assertEqual(self.data_operation.get_output(), {'output_files': []})
+    
+    def test_generate_cache_key(self):
+        cache_key = self.data_operation.generate_cache_key()
+        self.assertIsInstance(cache_key, str)
+        self.assertGreater(len(cache_key), 0)
+    
+    def test_set_get_status(self):
+        self.data_operation.set_status('COMPLETED')
+        self.assertEqual(self.data_operation.get_status(), 'COMPLETED')
+
+    def test_set_get_message(self):
+        self.data_operation.set_message('Test message')
+        self.assertEqual(self.data_operation.get_message(), 'Test message')
+    
+    def test_set_get_percent_completion(self):
+        self.data_operation.set_percent_completion(1.0)
+        self.assertEqual(self.data_operation.get_percent_completion(), 1.0)
+
+    def test_set_get_output(self):
+        self.data_operation.set_output({'output_files': []})
+        self.assertEqual(self.data_operation.get_percent_completion(), 1.0)
+        self.assertEqual(self.data_operation.get_status(), 'COMPLETED')
+        self.assertEqual(self.data_operation.get_output(), {'output_files': []})
+
+    def test_set_failed(self):
+        self.data_operation.set_failed('Test message')
+        self.assertEqual(self.data_operation.get_status(), 'FAILED')
+        self.assertEqual(self.data_operation.get_message(), 'Test message')
+
+class TestMedianOperation(TestCase):
+    
+    def testTrue(self):
```

Comment:
So in these classes is where I would expect the real meat of the tests to exist. You want to test that doing a median operation on two simple image arrays does what you expect, i.e. it generates a median of the inputs as an output.